gammapy/modeling/models/core.py

```diff
@@ -455,43 +455,186 @@ def copy(self):
-                )
+    def mask(
+        self,
+        name_substring=None,
+        datasets_names=None,
+        tag=None,
+        spatial_tag=None,
+        spectral_tag=None,
+        frozen=None,
+    ):
+        """Create a mask of models, true if all condition are verified
 
-            if tag:
-                selection &= tag in model.tag
+        Parameters
+        ----------
+        name_substring : str
+            Substring contained in the model name
+        dataset_name : str or list
+            Name of the dataset
+        tag : str or list
+            Model tag
+        spatial_tag : str or list
+            Spatial model tag if any
+        spectral_tag : str or list
+            Spectral model tag
+        frozen : bool
+            Select models with all parameters frozen if True, exclude them if False.
+ 
+       Returns
+        -------
+        mask : `numpy.array`
+            Boolean mask
+        """
+
+        selection = np.ones(len(self), dtype=bool)
 
+        for km, model in enumerate(self):
             if name_substring:
-                selection &= name_substring in model.name
+                selection[km] &= name_substring in model.name
+            if datasets_names:
+                if not isinstance(datasets_names, list):
+                    datasets_names = [datasets_names]
+                selection[km] &= model.datasets_names is None or np.any(
+                    [name in model.datasets_names for name in datasets_names]
+                )
+            if tag:
+                if not isinstance(tag, list):
+                    tag = [tag]
+                inlist = np.any([t in model.tag for t in tag])
+                selection[km] &= tag == model.tag or inlist
+            if spatial_tag:
+                if not isinstance(spatial_tag, list):
+                    spatial_tag = [spatial_tag]
+                if hasattr(model, "spatial_model"):
+                    selection[km] &= np.any(
+                        [t in model.spatial_model.tag for t in spatial_tag]
+                    )
+                else:
+                    selection[km] &= False
+            if spectral_tag:
+                if not isinstance(spectral_tag, list):
+                    spectral_tag = [spectral_tag]
+                if hasattr(model, "_spectral_model"):
+                    # use private attr to support BackgroundModel
+                    selection[km] &= np.any(
+                        [t in model._spectral_model.tag for t in spectral_tag]
+                    )
+                else:
+                    selection[km] &= False
+            if frozen is not None:
+                all_frozen = np.all([p.frozen for p in model.parameters])
+                if frozen == True:
+                    selection[km] &= all_frozen
+                else:
+                    selection[km] &= ~all_frozen
+        return selection
+
+    def set_parameters_bounds(
```

Comment:
It's not documented, but in general users are free to overwrite defaults of Gammapy models on the class level e.g.:

```python3
from gammay.modeling.models import PowerLawSpectralModel, Parameter

PowerLawSpectralModel.index = Parameter("index", value=3, min=0, max=5)
```
However I'm not sure it's pattern we should advertise, need to think about it once more....

Reply:
but set_parameters_bounds() can change the settings a posteriori

Reply:
Indeed, my first comment was not meant to replace a functionality, but just as an idea that came to my mind...